src/Aspire.Hosting.Orleans/OrleansService.cs

```diff
@@ -1,34 +1,59 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Aspire.Hosting.ApplicationModel;
+
 namespace Aspire.Hosting.Orleans;
 
 /// <summary>
 /// Describes an Orleans service.
 /// </summary>
-/// <param name="builder">The distributed application builder.</param>
-/// <param name="name">The service name.</param>
-public class OrleansService(IDistributedApplicationBuilder builder, string name)
+public class OrleansService
 {
+    /// <summary>Initializes a new <see cref="OrleansService"/> instance.</summary>
+    /// <param name="builder">The distributed application builder.</param>
+    /// <param name="name">The service name.</param>
+    public OrleansService(IDistributedApplicationBuilder builder, string name)
+    {
+        Name = name;
+        Builder = builder;
+        ServiceId = ParameterResourceBuilderExtensions.CreateGeneratedParameter(builder, $"{name}-service-id", secret: false, new GenerateParameterDefault
+        {
+            Lower = true,
+            Upper = false,
+            Special = false,
+            Numeric = true,
+            MinLength = 25
+        });
+        ClusterId = ParameterResourceBuilderExtensions.CreateGeneratedParameter(builder, $"{name}-cluster-id", secret: false, new GenerateParameterDefault
+        {
+            Lower = true,
+            Upper = false,
+            Special = false,
+            Numeric = true,
+            MinLength = 25
+        });
+    }
+
     /// <summary>
     /// Gets the name of the service.
     /// </summary>
-    public string Name { get; } = name;
+    public string Name { get; }
 
     /// <summary>
     /// Gets the distributed application builder.
     /// </summary>
-    public IDistributedApplicationBuilder Builder { get; } = builder;
+    public IDistributedApplicationBuilder Builder { get; }
 
     /// <summary>
     /// Gets or sets the service identifier.
     /// </summary>
-    public string? ServiceId { get; set; }
+    internal object ServiceId { get; set; }
```

Comment:
Why `object`?

```suggestion
    internal ParameterResource ServiceId { get; set; }
```

Reply:
It's either a parameter or a hard-coded string value.. I made it internal so at least this detail doesn't leak.

See `WithServiceId` and `WithClusterId`

Reply:
> It's either a parameter or a hard-coded string value.

Might be good sticking that comment in here.